src/BloomTests/Book/BookDataTests.cs

```diff
@@ -139,6 +139,18 @@ public void UpdateFieldsAndVariables_CustomLibraryVariable_CopiedToOtherElement(
 			Assert.AreEqual("aa", textarea2.InnerText);
 		}
 
+		[Test]
+		public void UpdateFieldsAndVariables_HasBookTitleTemplateWithVernacularPlaceholder_CreatesTitleForVernacular()
+		{
+			var dom = new HtmlDom(@"<html ><head></head><body>
+				<div id='bloomDataDiv'>
+						<div data-book='bookTitleTemplate' lang='{V}'>the title</div>
+				</div>
+				</body></html>");
+			var data = new BookData(dom, _collectionSettings, null);
+			data.UpdateVariablesAndDataDivThroughDOM();
+			AssertThatXmlIn.Dom(dom.RawDom).HasSpecifiedNumberOfMatchesForXpath("//div[@data-book='bookTitle' and @lang='"+_collectionSettings.Language1Iso639Code+"' and text()='the title']",1);
```

Comment:
Not for this issue, but seeing this reminded me again how careful one has to be with this method to note the exact number of matches expected. Almost the exact same pattern is used for "we expect to see this" and "we don't expect to see this". I think it would be helpful to reserve this method for unusual cases and introduce "HasOneMatchForXpath" and "DoesNotMatchXpath" for the two common ones.